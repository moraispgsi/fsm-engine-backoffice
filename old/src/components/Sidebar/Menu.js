/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import MenuItem from './MenuItem';

const StyledHeader = styled.li`
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  display: ${props => (props.collapse ? 'none !important' : 'list-item')};
  text-align: -webkit-match-parent;
  position: relative;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  padding: 10px 25px 10px 15px;
  font-size: 12px;
  background-attachment: scroll;
  background-clip: initial;
  background-image: none;
  background-origin: initial;
  background-position: initial;
  background-repeat: initial;
  background-size: initial;
  text-overflow: clip;

  /* collapse transform */
  ${props => props.collapse && '-webkit-transform: translateZ(0);'}

  /* theme */
  color: ${props => props.theme.sidebarHeaderColor || '#fff'}
  background-color: ${props => props.theme.sidebarHeaderBg || '#fff'}
`;

const StyledMenu = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  overflow: hidden;

  &:hover {
    overflow: visible;
  }
`;

const renderChildren = (children, sidebarCollapse) => (
  React.Children.map(children, child =>
    React.cloneElement(child, {
      collapse: sidebarCollapse,
    }),
  )
);

const Menu = ({ children, header, collapse }) => (
  <StyledMenu collapse={collapse} >
    <StyledHeader collapse={collapse} >{header}</StyledHeader>
    {renderChildren(children, collapse)}
  </StyledMenu>
);

Menu.propTypes = {
  children: React.PropTypes.node,
  header: React.PropTypes.string,
  collapse: React.PropTypes.bool,
};

Menu.Item = MenuItem;

export default Menu;
