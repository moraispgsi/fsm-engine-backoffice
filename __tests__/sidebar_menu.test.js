/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import red from '../src/styles/skin-red';
import Menu from '../src/components/Sidebar/Menu';

describe('<Menu />', () => {
  it('renders correctly with default options', () => {
    const component = renderer.create(<Menu />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with theme', () => {
    const component = renderer.create(<ThemeProvider theme={red}>
      <Menu />
    </ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with header', () => {
    const component = renderer.create(<Menu header="A Header" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with collapse option', () => {
    const component = renderer.create(<Menu collapse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const component = renderer.create(<Menu >
      <span>Child</span>
    </Menu>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
