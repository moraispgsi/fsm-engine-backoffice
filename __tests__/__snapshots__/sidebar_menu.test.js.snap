exports[`<Menu /> renders children correctly 1`] = `
<ul
  className="jHSOkj">
  <li
    className="hhsfIK" />
  <span
    collapse={undefined}>
    Child
  </span>
</ul>
`;

exports[`<Menu /> renders correctly with collapse option 1`] = `
<ul
  className="jHSOkj">
  <li
    className="gaPufh" />
</ul>
`;

exports[`<Menu /> renders correctly with default options 1`] = `
<ul
  className="jHSOkj">
  <li
    className="hhsfIK" />
</ul>
`;

exports[`<Menu /> renders correctly with header 1`] = `
<ul
  className="jHSOkj">
  <li
    className="hhsfIK">
    A Header
  </li>
</ul>
`;

exports[`<Menu /> renders correctly with theme 1`] = `
<ul
  className="jHSOkj">
  <li
    className="cCzeVt" />
</ul>
`;
